File: BOJ/탐색과정렬/10828스택.py
import sys
input = sys.stdin.readline
stack = []
n = int(input())
for _ in range(n):
    a = input()
    a = a[:-1]
    
    if a.split()[0] == 'push':
        stack.append(a.split()[1])
    elif 'pop' == a:
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())
    elif 'size' == a:
        print(len(stack))
    elif 'empty' == a:
        if len(stack) == 0 :
            print(1)
        else:
            print(0)
    elif 'top' == a:
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])

File: BOJ/자료구조/10828_스택.cpp
#include <iostream>
#include <stack> 
#include <string> 
using namespace std;
int main(void) {
	ios::sync_with_stdio(0);
	cin.tie(0);
	stack<int> s;
	int n;
	cin>> n;
	while(n--){
		string c;
		cin >> c;
		if(c == "push"){
			int a;
			cin>>a;
			s.push(a);
		} 
		else if(c == "top"){
			if(s.empty()) cout<< -1<<'\n';
			else cout<< s.top()<< '\n';
		}
		else if(c == "pop"){
			if(s.empty()) cout<< -1<<'\n';
			else{
				cout<< s.top() <<'\n';
				s.pop()
;			} 
			
		}
		else if(c == "size"){
			cout << s.size()<<'\n';
		}
		else{
			cout << (int)s.empty() << '\n';
		}
	}
}
